.github/workflows/test.yaml

```
@@ -26,7 +26,7 @@ jobs:
       fail-fast: false
       matrix:
         # Run windows only on scheduled runs on Sundays, otherwise ignore
-        os: ${{ github.event.schedule == '0 5 * * 0' && fromJson('["ubuntu", "macos", "windows"]') || fromJson('["ubuntu", "macos"]') }}
+        os: ${{ github.event.schedule == '0 5 * * 0' && fromJson('["ubuntu", "macos", "windows"]') || github.ref_name == 'releases/**' && fromJson('["ubuntu", "macos", "windows"]') || fromJson('["ubuntu"]') }}
```

So this would be the suggested approach of running it weekly on the schedule analogue to the Windows CI.
@siddharth-krishna I am not sure what is needed as a setting though, so that we can manually run the macOS CI on release branches. Do you know? 
```suggestion
        os: ${{ github.event.schedule == '0 5 * * 0' && fromJson('["ubuntu", "macos", "windows"]') || fromJson('["ubuntu"]') }}
```

---

I like the second suggestion!